app/src/main/java/org/desarrolladorslp/teckersapp/ui/messages/MessageViewModel.kt

```diff
@@ -1,18 +1,36 @@
 package org.desarrolladorslp.teckersapp.ui.messages
 
-import androidx.lifecycle.LiveData
 import androidx.lifecycle.MutableLiveData
-import androidx.lifecycle.Transformations
 import androidx.lifecycle.ViewModel
+import org.desarrolladorslp.teckersapp.model.Inbox
+import org.desarrolladorslp.teckersapp.model.MessageHeader
+import org.desarrolladorslp.teckersapp.service.APIEndpoint
+import org.desarrolladorslp.teckersapp.service.MessageService
+import org.desarrolladorslp.teckersapp.service.NetworkCall
+
 
 class MessageViewModel : ViewModel() {
 
-    private val _index = MutableLiveData<Int>()
-    val text: LiveData<String> = Transformations.map(_index) {
-        "Hello world from section: $it"
-    }
+        private var messageService = APIEndpoint.instance()?.create(MessageService::class.java);
+        private val _inbox = MutableLiveData<Inbox>().apply {
+
+            value=generateInbox().value!!.data!!
+
+        }
+        val inbox: Inbox = _inbox.value!!
+
+        fun generateInbox() = NetworkCall<Inbox>().makeCall(messageService!!.getMessages())
```

Comment:
Now that you are getting the actual inbox, it would be a good idea to rename this method

Reply:
it could be `getInbox()`